File: Actividad21CA/app/src/main/java/mx/itesm/actividad21ca/MainActivity.java
package mx.itesm.actividad21ca;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements Handler.Callback, View.OnClickListener {
    private Handler handler;
    private RecyclerView recyclerView;
    private ArrayList<JSONObject> data;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        /*Rquest*/
        handler = new Handler(Looper.getMainLooper(), this);

        /*Recycler*/
        data = new ArrayList<>();

    }

    public void doRequest(View v){
        // checar de manera local
        Request request = new Request("https://raw.githubusercontent.com/CArmenta5/Moviles/master/Friend.json", handler);
        request.start();


    }

    @Override
    public boolean handleMessage(@NonNull Message msg) {
        String respuesta = msg.obj.toString();
        try {
            JSONArray dataFriends = new JSONArray(respuesta);
            for(int i = 0; i < dataFriends.length(); i++){
                JSONObject friend = dataFriends.getJSONObject(i);
                data.add(friend);
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }
        recyclerView = findViewById(R.id.recyclerView);
        FriendAdapter adapter = new FriendAdapter(data,this);

        LinearLayoutManager llm = new LinearLayoutManager(this);
        llm.setOrientation(LinearLayoutManager.VERTICAL);

        recyclerView.setLayoutManager(llm);
        recyclerView.setAdapter(adapter);


        return true;

    }

    @Override
    public void onClick(View v) {
        int pos = recyclerView.getChildLayoutPosition(v);

        try {
            Toast.makeText(this,
                    "Friend: " +
                            data.get(pos).getString("nombre")+", "+
                            data.get(pos).getString("hobby")+", "+
                            data.get(pos).getInt("age")+", "+
                            data.get(pos).getInt("phone")+", "+
                            data.get(pos).getString("address")+" "
                    , Toast.LENGTH_SHORT).show();
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }
}

File: Actividad22CA/app/src/main/java/mx/itesm/actividad22ca/MainActivity.kt
package mx.itesm.actividad22ca

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject
import kotlin.collections.ArrayList

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private var recyclerView: RecyclerView? = null
    private var data: ArrayList<JSONObject>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        /*Inicialización atributos*/
        data = ArrayList()
        recyclerView =  findViewById(R.id.recyclerView)
        val json = "[{'name':'burger', 'price':15, 'description':'a juicy burger!'},{'name':'hotdog', 'price':20, 'description':'an ok hot dog'},{'name':'tacos', 'price':12, 'description':'some pretty good tacos'},{'name':'torta', 'price':22, 'description':'nice torta'},{'name':'carne asada', 'price':50, 'description':'a great carne asada'}]"

        /*RecyclerView*/
        val menu  = JSONArray(json)
        for (i in 0 until menu.length()) {
            val platillo = menu.getJSONObject(i)
            data?.add(platillo)
        }

        val adapter = Adapter(data?: ArrayList(), this)
        recyclerView?.layoutManager = LinearLayoutManager(this,LinearLayoutManager.VERTICAL ,false)
        recyclerView?.adapter = adapter

    }
    override fun onClick(v: View?) {
        val pos: Int = v?.let { recyclerView?.getChildLayoutPosition(it) } ?: 0
        try {
            Toast.makeText(
                this,
                "Description: " +
                        (data?.get(pos)?.getString("description") ?: "")
                , Toast.LENGTH_SHORT
            ).show()
        } catch (e: JSONException) {
            e.printStackTrace()
        }
    }
}